pkg/apis/crd/v1alpha1/types.go

```diff
@@ -325,3 +325,102 @@ type NodeLatencyMonitorList struct {
 
 	Items []NodeLatencyMonitor `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BGPPolicy defines BGP configuration applied to Nodes.
+type BGPPolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard metadata of the object.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec BGPPolicySpec `json:"spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type BGPPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []BGPPolicy `json:"items"`
+}
+
+// BGPPolicySpec defines the specification for a BGPPolicy.
+type BGPPolicySpec struct {
+	// NodeSelector selects Nodes to which the BGPPolicy is applied. If multiple BGPPolicies select a Node, only one
+	// will be effective and enforced; others serve as alternatives.
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector"`
+
+	// LocalASN is the AS number used by the BGP process.
+	LocalASN *int32 `json:"localASN,omitempty"`
+
+	// ListenPort is the port on which the BGP process listens.
+	ListenPort *int32 `json:"listenPort,omitempty"`
+
+	// Advertisements selects IPs or CIDRs to be advertised to BGP peers.
+	Advertisements *Advertisements `json:"advertisements,omitempty"`
+
+	// BGPPeers is the list of BGP peers.
+	BGPPeers []BGPPeer `json:"bgpPeers,omitempty"`
+}
+
+type Advertisements struct {
+	// Determine how to advertise Service IPs of the selected Services if this is set. It's worth noting that if all the
+	// flags (ClusterIPs, ExternalIPs and LoadBalancerIPs) in ServiceAdvertisement are false or nil, no Service IP will
+	// be advertised.
+	Service *ServiceAdvertisement `json:"service,omitempty"`
+
+	// Advertise NodeIPAM Pod CIDRs of the selected Nodes if set.
+	Pod *PodAdvertisement `json:"pod,omitempty"`
+
+	// Advertise Egress IPs assigned to the selected Nodes if set.
+	Egress *EgressAdvertisement `json:"egress,omitempty"`
+}
+
+type ServiceAdvertisement struct {
+	// Determine whether to advertise ClusterIPs of the selected Services (currently all Services without Service selector).
+	ClusterIPs *bool `json:"clusterIPs,omitempty"`
+
+	// Determine whether to advertise ExternalIPs of the selected Services (currently all Services without Service selector).
+	ExternalIPs *bool `json:"externalIPs,omitempty"`
+
+	// Determine whether to advertise LoadBalancerIPs of the selected Services (currently all Services without Service selector).
+	LoadBalancerIPs *bool `json:"loadBalancerIPs,omitempty"`
```

Comment:
ditto, it's unclear whether they will be advertised when not set by looking at these fields. I think you could remove "It's worth noting that ..." at L374 and just explain each field above itself.

Reply:
Will do.